pageserver/src/tenant.rs

```diff
@@ -3988,6 +4023,13 @@ pub(crate) mod harness {
         buf.freeze()
     }
 
+    #[cfg(test)]
+    pub async fn advance_time_by(duration: Duration) {
```

Comment:
I would be against this method; the pausedness state of tokio is unqueryable, thus we should not manage it.

Tests can (and usually do) start paused if they are to manipulate the timer.

Instead we should just inline this to all tests which need it.

Reply:
Hmm at least `test_prohibit_branch_creation_on_garbage_collected_data` could be `#[tokio::test(start_paused = true)]` and then you could start the test by sleeping for the lsn lease time.

Then it would be more obvious that this test is such that manipulates time.

I think similarly for the other test.

Reply:
Did as you suggested: 01fce43f1c3c54a3878dfe0e81b462e7ce27b6da

Reply:
Also refactored `test_lsn_lease` in 0c0bfd4d912aa92937c1b6b463e4b1294558b0fd